jobs/jobs/schemas.py

```diff
@@ -343,3 +346,27 @@ class JobProgress(BaseModel):
     finished: int = Field(..., example=1)
     total: int = Field(..., example=1)
     mode: str = Field(..., example="Automatic")
+
+
+class PipelineEngine(BaseModel):
+    name: str
+    resource: str
+    enabled: bool
+
+
+class PipelineEngineSupport(BaseModel):
+    data: List[PipelineEngine]
+
+
+class Pipeline(BaseModel):
+    id: str
+    name: str
+    version: int = Field(default=0)
+    type: str = Field(default="airflow")
+    date: datetime
+    meta: Dict[str, Any]
+    steps: List[Any] = Field(default_factory=list)
```

Comment:
BTW Pydantic handles default empty list correctly, without non-hashable object side effects.
```suggestion
    steps: List[Any] = []
```